### DISCUSSION TITLE
2024-08-09 금요일

### DISCUSSION THREAD

**Discussion by eunohhh** [nbc-final-a10/TripBuddy#247] [기록보드]
2024-08-09 금요일 기록보드입니다.

## 여행가고십조(A10) 일일 공통 일과표

| 시간 | 일과 |  기록 |
| :-: | :-: | :-: |
| 09:00 ~ 10:00 | 아침 회의 | |
| 10:00 ~ 11:00 | 기술면접 준비 | |
| 11:00 ~ 13:00 | 집중 개발 time | |
| 13:00 ~ 14:00 | 점심 식사 | |
| 14:00 ~ 18:00 | 집중 개발 time | |
| 18:00 ~ 19:00 | 저녁 식사 | |
| 19:00 ~ 20:00 | 저녁 회의 | |
| 20:00 ~ 21:00 | 5분 기록보드 작성 / 개인 회고 | |

```
## 개인 기록보드 서식 (김병준)

### 오늘 한 일

- 내용

### 트러블 슈팅

- 내용

### 내일 계획

- 내용
```

```
// 이미지 테이블
<table>
  <tr>
    <th>이미지 1번 제목</th>
    <th>이미지 2번 제목</th>
  </tr>
  <tr>
    <td><img src="이미지 주소" width="300" height="300"/></td>
    <td><img src="이미지 주소" width="300" height="300"/></td>
  </tr>
</table>
```

**Comment by eunohhh** [comment]
## 개인 기록보드 서식 (오은)

### 오늘 한 일

- [x] : 스토리 글쓰기 시 옵션 조금 상세하게 
- [x] : 튜토리얼 데스크탑에서 헤더 숨기기 
- [x] : pwa 연구 - 하단 참조(ios에서는 불가능)
- [x] : 스토리 삭제 
- [x] : 스토리 좋아요 
- [x] : 스토리 좋아요 옵티미스틱 
- [x] : 모바일에서 스토리 인풋문제 
- [x] : 홈화면 카드에서도 스토리 좋아요 노출하기
- [x] : 네이버 검수요청 
- [x] : 미들웨어 설정
- [x] : 여정 데스크탑 페이지네이션 
- [x] : 여정 관련 UI 수정작업
- [x] : 여정 모바일 무한스크롤
- [x] : 스토리 데탑/모바일 위치 맞추기
- [x] : 데스크탑 홈에서 마우스 스크롤 동작 수정 - 세로스크롤 가능하게하고, 화살표로 목록 이동하고, 좌우스크롤은 옵션
- [x] : error.js not-found.js 등 없는 기능들 
- [x] : 홈배너 로직 수정 - 탠스택쿼리
- [x] : 메인 스테일타임 점검 - 5분으로 설정
- [x] : 하단바 svgr 및 fixed
- [x] : 프로필 에디트 라우트 이동
- [ ] : 여정 카드에 플필사진 4개까지 보이게끔
- [ ] : 여정 찜하기 
- [ ] : 글수정 페이지 만들기


### 트러블 슈팅

- pwa 설치 유도 문구를 ios 에서도 보여지게 하기 위해 navigator.share 를 사용하였으나 되지 않는 것을 최종 확인

### 내일 계획

- 찜하기 만들고, 글수정페이지 작성 시작하기

**Comment by Captain-Kim** [comment]
## 개인 기록보드 서식 (김병준)

### 오늘 한 일

- 리치텍스트 에디터 구현
- 버디프로필 필러우 기능 버그 픽스 (생애주기 재설정)

### 트러블 슈팅

- 컴포넌트의 렌더링 순서를 제대로 고려하지 못해 follwoingId가 간헐적으로 불러와지지 않는 문제가 발생함.

```
const followingId = window.location.pathname.split('/').pop();
```

- 위와 같은 형태로 window 객체를 활용하여 followingId를 url parameter에서 추출하고 있었음.
- 서버 사이드 렌더링에서는 window 객체가 존재하지 않기 때문에 리액트의 초기 렌더링 상태에서는 window 객체에 접근하지 못해 값이 undefined 로 초기화 될 수 있다.
- 리액트는 클라이언트 사이드 렌더링이 완료된 후에야 브라우저에서 window 객체에 접근할 수 있다.
- 문제는 window 객체의 실행 순서를 몰라서 위 코드를 코드 최상단에서 실행하고 있었는데, 나는 코드 최상단에서 실행하면 동기적으로 실행되며 우선되어 실행될 것으로 예상했다.
- useEffect 훅 내부로 window 객체를 옮기고 나서 문제가 해결되었음.
- useEffect 훅은 리액트 컴포넌트가 브라우저에서 렌더링 된 후에 실행되기 때문에 window 객체에 안전하게 접근할 수 있도록 그 순서를 보장하기 때문에 해결 된 사례임.

### 내일 계획

- 리치 텍스트 에디터 구현
